test/unit/test_intrahost.py

```diff
@@ -471,6 +471,24 @@ def test_backfill_sample_from_assembly(self):
         self.assertEqual(':'.join(rows[0][1].split(':')[:2]), '1:1.0')
         self.assertEqual(rows[0][2], '.:.:.:.:.')
 
+    def test_output_all_positions(self):
+        # If full output is enabled, check that we output consensus values
+        # REF C
+        # S1  A (isnv)
+        # S2  A (consensus, no isnv)
+        merger = VcfMergeRunner([('ref1', 'ATCG')])
+        merger.add_genome('s1', [('s1-1', 'ATCC')])
+        merger.add_genome('s2', [('s2-1', 'ATAG')])
+        merger.add_snp('s1', 's1-1', 3, [('C', 90, 90), ('A', 10, 10)])
+        rows = merger.run_and_get_vcf_rows(output_all_positions=True)
+        self.assertEqual(len(rows), 4)
+        self.assertEqual(rows[3].contig, 'ref1')
+        self.assertEqual(rows[3].pos + 1, 4)
+        self.assertEqual(rows[3].ref, 'G')
+        self.assertEqual(rows[3].alt, 'C')
+        self.assertEqual(':'.join(rows[3][0].split(':')[:2]), '1:1.0')
+        self.assertEqual(':'.join(rows[3][1].split(':')[:2]), '0:0.0')
```

Comment:
This should maybe be `self.assertEqual(':'.join(rows[3][1].split(':')[:2]), '0:.')` for positions we are imputing from ref, since absence of iSNVs in the vphaser output does not necessarily mean iSNVs were absent, only that they did not warrant inclusion in the output.